web/Controllers/LandlordController.cs

```diff
@@ -53,24 +59,50 @@ public async Task<ActionResult> RegisterPost([FromForm] LandlordProfileModel cre
         }
 
         var user = GetCurrentUser();
+        var result = ILandlordService.LandlordRegistrationResult.ErrorUserAlreadyHasLandlordRecord;
+
+        if (createModel.Unassigned)
+        {
+            result =
+                await _landlordService.RegisterLandlord(createModel);
+        }
+        else
+        {
+            result =
+                await _landlordService.RegisterLandlord(createModel,
+                    user);
+        }
 
-        var result = await _landlordService.RegisterLandlordWithUser(createModel, user);
 
         switch (result)
         {
             case ILandlordService.LandlordRegistrationResult.Success:
-                _logger.LogInformation("Successfully created landlord for user {UserId}", user.Id);
-                return Redirect(Url.Action("MyProfile")!);
+                _logger.LogInformation("Successfully created landlord for user");
+                if (!createModel.Unassigned)
+                {
+                    return Redirect(Url.Action("MyProfile")!);
+                }
+                else
+                {
+                    return RedirectToAction("LandlordList", "Admin");
+                }
 
             case ILandlordService.LandlordRegistrationResult.ErrorLandlordEmailAlreadyRegistered:
                 _logger.LogWarning("Email already registered {Email}", createModel.Email);
                 ModelState.AddModelError("Email", "Email already registered");
-                return View("Register");
+                return View("Register", createModel);
 
             case ILandlordService.LandlordRegistrationResult.ErrorUserAlreadyHasLandlordRecord:
                 _logger.LogWarning("User {UserId} already associated with landlord", user.Id);
                 TempData["FlashMessage"] = "Already registered!"; // This will be displayed on the Profile page
-                return Redirect(Url.Action("MyProfile")!);
+                if (!createModel.Unassigned)
+                {
+                    return Redirect(Url.Action("MyProfile")!);
+                }
+                else
+                {
+                    return RedirectToAction("LandlordList", "Admin");

```

Comment:
This should never happen right? I wouldn't bother handling it, the original behaviour of redirecting to the profile with a flash message seems fine.